### PR TITLE
Graph representation

### LINKED ISSUE #35: Simplify graph representation
Currently our graph representation is very rudimentary. It takes the nodes and ways of an OSM file and simply creates edges between nodes according to the sequence in  the `way` elements. The problem with this approach is that it is very inefficient since lots of nodes  exists only to represent geographical coordinates. 

A better approach would be one like the one used in OSRM: https://github.com/Project-OSRM/osrm-backend/wiki/Graph-representation

Although I dont completely agree with the fully expanded graph representation that they use. I think that simplifying the connections between nodes would greatly improve the current algorithm both in speed and memory. It has the added benefit that it would be easier to couple with the changes necessary for #28 since geo-coordinates dont need network metadata as opposed to nodes in the graph.

### PR DESCRIPTION
**Pull request by carocad**
- simplify Valuable protocol by requiring only cost parameter instead of both cost and time
- SimpleValue type removed in favor of Number
- rollback: out-arcs and in-arcs as hash-maps
- GeoCoordinate protocol added

fixes #35

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -23,6 +23,6 @@
             [lein-ring "0.12.0"]]
   :ring {:handler service.routing.core/app
          :auto-reload? true}
-  :jvm-opts ["-Xmx300m"])
+  :jvm-opts ["-Xmx500m"])
```
Comment (mehdisadeghi) [inline project.clj]: Should we also defined the minimum required memory using `Xms` flag?

Comment (carocad) [inline project.clj]: @mehdisadeghi I dont think we would gain anything from that. In general I think we should make it as small as possible so I dont know if we would gain anything by defining the minimum size.

**Comment by carocad** [comment]
@mehdisadeghi The PR is still in progress so dont worry. I will make some more changes and then you can review it.

As always, dont worry about not being able to review it, but rather if you have any doubts about what the code is doing or why is it doing it that way this is the place to start.

I want to keep you on the loop :)

**Comment by carocad** [comment]
now it should also:
- fixes #27 
- fixes #28

**Comment by carocad** [comment]
@mehdisadeghi now this is ready for review :)

The changes are (not in order):
- simplify Valuable protocol by requiring only cost parameter instead of both cost and time
- SimpleValue type removed in favor of Number
-  rollback: arcs as hash-maps of their destination
- change arcs representation. Now we represent the pedestrian graph as an undirected graph. Since people can walk in any direction regardless of the signs. The traversal is done is a directed way though i.e. both `predecesors` and `sucessors` still exists and return the incoming and outgoing arcs respectively
- GeoCoordinate protocol added to represent points in space that are not needed to represent the graph
- the OSM reading function now returns a (network) map with :graph :ways :points. There is an explanation about them in the code ;)
- only import the OSM nodes and ways that are related to pedestrian routing

One side effect of all these changes is that now we have to read the OSM file twice for building up (what I know call) the `network`. The problem with the previous approach is that it consumes too much memory as we need to keep too many elements in memory before we can filter out the ones that we are actually interested in. There are some comparisons in the osm file.

**Comment by carocad** [comment]
It seems that this also breaks on the server side. I will do some more tests to get it working

**Pull request merged**